x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/saml/SamlLogoutResponseHandler.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.security.authc.saml;
+
+import org.elasticsearch.common.unit.TimeValue;
+import org.opensaml.saml.saml2.core.LogoutResponse;
+import org.w3c.dom.Element;
+
+import java.time.Clock;
+import java.util.Collection;
+
+import static org.elasticsearch.xpack.security.authc.saml.SamlUtils.samlException;
+
+public class SamlLogoutResponseHandler extends SamlResponseHandler {
+
+    private static final String LOGOUT_RESPONSE_TAG_NAME = "LogoutResponse";
+
+    public SamlLogoutResponseHandler(
+        Clock clock, IdpConfiguration idp, SpConfiguration sp, TimeValue maxSkew) {
+        super(clock, idp, sp, maxSkew);
+    }
+
+    public void handle(String content, Collection<String> allowedSamlRequestIds) {
+        final ParsedQueryString parsed = parseQueryStringAndValidateSignature(content, "SAMLResponse");
+        final Element root;
+        if (parsed.hasSignature) {
```

Comment:
I would very much prefer that we are explicit here and not try to deduce and imply checks. I think we need two parameters for the REST layer, one for the body of a response that comes via the HTTP-POST binding and a _different_ one for the query string of a response that comes via the HTTP-Redirect binding. Then we can be very explicit here with regards to what we expect to have and what we need to process/check/validate

Reply:
See above.